osu.Game/Overlays/Comments/OnlineCommentsContainer.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Bindables;
+using osu.Game.Online.API.Requests;
+using osu.Game.Users;
+using System.Threading.Tasks;
+
+namespace osu.Game.Overlays.Comments
+{
+    public class OnlineCommentsContainer : CommentsContainer
+    {
+        private GetCommentsRequest request;
+        private int currentPage;
+        private CommentBundleParameters parameters;
+
+        public void ShowComments(CommentableType type, long id)
+        {
+            parameters = new CommentBundleParameters(type, id);
+            Sort.TriggerChange();
```

Comment:
The way this class is implemented is pretty weird. You're doing calls on members of the base class that seem like they should go in the base class themselves. Why is this done here?

Furthermore, why are the implementations of `OnSortChanged()`, `OnUserChanged()` and `OnLoadStarted()` not in the base class?

This all may be simplified via a better implementation of the retrieval in the first place. Consider looking at `Leaderboard` as a reference of the implementation of comment retrieval. In general - expose a `FetchComments()` method, and do the success/cancelling stuff in the base class.

You'll also find that you may not need to expose the methods listed above through this refactoring.

Reply:
We can't do much with sort for "offline mode" (when we only have a local bundle)
The overall idea was just move all the online logic outside of the class itself (in tjis case to it's sub-class)

Reply:
> Furthermore, why are the implementations of OnSortChanged(), OnUserChanged() and OnLoadStarted() not in the base class?
`OnSortChanged` do nothing in the base class when it fetches new Bundle in Online one
`OnUserChanged` reloads the same comment bundle in the base class and refetches new Bundle in Online one
`OnLoadStarted` in the `OnlineCommentsContainer` is just an expansion (as you can see by using `base.OnLoadStarted`)
> do the success/cancelling stuff in the base class.
The overall idea was just move all the online logic outside of the class itself (in this case to it's sub-class)

Reply:
Well all comments are online comments. The only reason the base class needs to be provided without online functionality is for testing purposes (although in the future even that will be arguable).
So I'd like to see the logic unified - always refetch when sort is changed, always refetch when the user is changed, always cancel request / reset state in `OnLoadStarted()`.

Reply:
Okay, I'll remove separated `OnlineCommentsContainer` and will make a single one again

Reply:
I don't think merging the containers back again is necessary for what @smoogipoo wants to be done here, if I understood correctly.

Reply:
Correct, I just want the logic to be standardised. The online one can still exist for performing requests, but that's _all_ it should be doing, and exposing the requests to the base class via APIRequest or similar (again, look at `Leaderboard` for an example).

Reply:
Should be better now. I tried to mimic `Leaderboard` logic